raiden_contracts/tests/test_channel_cooperative_settle.py

```diff
@@ -7,147 +7,202 @@
+    )
+    (signature_A2, signature_B2) = create_withdraw_signatures(
+        [A, B], channel_identifier, B, balance_B, expiration
     )
 
+    # negative balances
     with pytest.raises(ValidationError):
         token_network.functions.cooperativeSettle(
-            channel_identifier, A, -1, B, balance_B, signature_A, signature_B
+            channel_identifier,
+            (A, -1, expiration, signature_A1, signature_B1),
+            (B, balance_B, expiration, signature_B2, signature_A2),
         )
     with pytest.raises(ValidationError):
         token_network.functions.cooperativeSettle(
-            channel_identifier, A, balance_A, B, -1, signature_A, signature_B
+            channel_identifier,
+            (A, balance_A, expiration, signature_A1, signature_B1),
+            (B, -1, expiration, signature_B2, signature_A2),
         )
+    # wrong participant addresses
     with pytest.raises(ValidationError):
         token_network.functions.cooperativeSettle(
-            channel_identifier, 0x0, balance_A, B, balance_B, signature_A, signature_B
+            channel_identifier,
+            (0x0, balance_A, expiration, signature_A1, signature_B1),
+            (B, balance_B, expiration, signature_B2, signature_A2),
         )
     with pytest.raises(ValidationError):
         token_network.functions.cooperativeSettle(
-            channel_identifier, A, balance_A, 0x0, balance_B, signature_A, signature_B
+            channel_identifier,
+            (A, balance_A, expiration, signature_A1, signature_B1),
+            (0x0, balance_B, expiration, signature_B2, signature_A2),
         )
+    # invalid signatures
     with pytest.raises(ValidationError):
         token_network.functions.cooperativeSettle(
-            channel_identifier, A, balance_A, B, balance_B, 0x0, signature_B
+            channel_identifier,
+            (A, balance_A, expiration, 0x0, signature_B1),
+            (B, balance_B, expiration, signature_B2, signature_A2),
         )
     with pytest.raises(ValidationError):
         token_network.functions.cooperativeSettle(
-            channel_identifier, A, balance_A, B, balance_B, signature_A, 0x0
+            channel_identifier,
+            (A, balance_A, expiration, signature_A1, 0x0),
+            (B, balance_B, expiration, signature_B2, signature_A2),
         )
-
+    with pytest.raises(ValidationError):
+        token_network.functions.cooperativeSettle(
+            channel_identifier,
+            (A, balance_A, expiration, signature_A1, signature_B1),
+            (B, balance_B, expiration, 0x0, signature_A2),
+        )
+    with pytest.raises(ValidationError):
+        token_network.functions.cooperativeSettle(
+            channel_identifier,
+            (A, balance_A, expiration, signature_A1, signature_B1),
+            (B, balance_B, expiration, signature_B2, 0x0),
+        )
+    # empty addresses
     with pytest.raises(TransactionFailed):
```

Comment:
I like to check for the specific error with
```
with pytest.raises(TransactionFailed, match="error message"):
```
whenever possible, so that we know we are actually exercising the correct require. That does require error message, of course. But we should take the opportunity to add error messages (https://github.com/raiden-network/raiden-contracts/issues/81) before the next mainnet deployment now that #401 is done.

Reply:
Done for all tests where transaction are expected to fail. I didn't add it to the `web3` formatting tests. Do you think it's useful there as well?

Reply:
I don't think so. IMO, those have limited value, anyway.